src/main/scala/PopModel.scala

```diff
@@ -19,130 +19,194 @@ package org.template
     // end should always be 'now' except in unusual conditions like for testing
-    val end = if (offsetDate.isEmpty ) DateTime.now else {
+    val end = if (offsetDate.isEmpty) DateTime.now else {
       try {
         ISODateTimeFormat.dateTimeParser().parseDateTime(offsetDate.get)
       } catch {
-        case e: IllegalArgumentException => e
+        case e: IllegalArgumentException =>
           logger.warn("Bad end for popModel: " + offsetDate.get + " using 'now'")
           DateTime.now
       }
     }
 
+    val interval = new Interval(end.minusSeconds(duration), end)
+
     // based on type of popularity model return a set of (item-id, ranking-number) for all items
-    logger.info(s"PopModel ${modelName} using end: ${end}, and duration: ${duration} ")
+    logger.info(s"PopModel $modelName using end: $end, and duration: $duration, interval: $interval")
+
+    // if None? debatable, this is either an error or may need to default to popular, why call popModel otherwise
     modelName match {
-      case Some("popular") => calcPopular(appName, eventNames, new Interval(end.minusSeconds(duration), end))
-      case Some("trending") => calcTrending(appName, eventNames, new Interval(end.minusSeconds(duration), end))
-      case Some("hot") => calcHot(appName, eventNames, new Interval(end.minusSeconds(duration), end))
-      case _ => None // debatable, this is either an error or may need to default to popular, why call popModel otherwise
+      case RankingType.Popular     => calcPopular(appName, eventNames, interval)
+      case RankingType.Trending    => calcTrending(appName, eventNames, interval)
+      case RankingType.Hot         => calcHot(appName, eventNames, interval)
+      case RankingType.Random      => calcRandom(appName, interval)
+      case RankingType.UserDefined => sc.emptyRDD
+      case unknownRankingType =>
+        logger.warn(
+          s"""
+             |Bad rankings param type=[$unknownRankingType] in engine definition params, possibly a bad json value.
+             |Use one of the available parameter values ($RankingType).""".stripMargin)
+        sc.emptyRDD
     }
+
   }
 
-  /** Creates a rank from the number of named events per item for the duration */
-  def calcPopular(appName: String, eventNames: List[String] = List.empty,
-    interval: Interval)(implicit sc: SparkContext): Option[RDD[(String, Float)]] = {
+  /** Create random rank for all items */
+  def calcRandom(
+    appName: String,
+    interval: Interval): RDD[(ItemID, Double)] = {
+
+    val events = eventsRDD(appName = appName, interval = interval)
+    val actionsRDD = events.map(_.targetEntityId).filter(_.isDefined).map(_.get).distinct()
+    val itemsRDD = fieldsRDD.map { case (itemID, _) => itemID }
+
+    //    logger.debug(s"ActionsRDD: ${actionsRDD.take(25).mkString(", ")}")
+    //    logger.debug(s"ItemsRDD: ${itemsRDD.take(25).mkString(", ")}")
+    actionsRDD.union(itemsRDD).distinct().map { itemID => itemID -> Random.nextDouble() }
```

Comment:
Generate random rank for all items. 
`itemsRDD` from `DataSource`, all items whitch have $set
`actionsRDD` - all items whitch have action